arches/app/models/models.py

```diff
@@ -523,20 +523,149 @@ def is_editable(self):
         else:
             return True
 
-    def get_published_graph(self, language=None, raise_if_missing=False):
+    def should_use_published_graph(self):
+        return bool(
+            self.publication
+            and not self.source_identifier
+            and not self.has_unpublished_changes
+        )
+
+    def get_published_graph(self, language=None):
         if not language:
             language = translation.get_language()
 
-        try:
-            graph = PublishedGraph.objects.get(
-                publication=self.publication, language=language
+        published_graph = PublishedGraph.objects.filter(
+            publication=self.publication, language=language
+        ).first()
+
+        return published_graph
```

Comment:
This eagerly does a query. I try to avoid starting from another manager. If you start with `self`, you make it possible for the caller to prefetch related objects and have that used here.

```suggestion
        if not self.publication_id:
            return None
        
        return self.publication.publishedgraph_set.filter(language=language).first()
```

Now, my example about prefetching doesn't totally work because of the filter() and the first(). You can return from a loop to handle that. Trade fetching a couple extra rows for the ability to avoid later fetches on subsequent calls (and we are calling this over and over -- so it might be worth being a little picky.)

```suggestion
        if not self.publication_id:
            return None
        
        for published_graph in self.publication.publishedgraph_set.all():
            if published_graph.language_id == language:
                return published_graph
        return None
```

This won't just help picky callers who are in love with `prefetch_related()` -- looks like there's motion in Django 6 to get "prefetching by default" as a settings bit 🥳